acquisitions/src/main/resources/thunderjet/mod-consortia/features/user-tenant-associations.feature

```diff
@@ -4,14 +4,484 @@ Feature: Consortia User Tenant associations api tests
+    When method GET
+    Then status 200
+
+    * def fun = function(user) {return  user.id == consortiaSystemUserInUniversityId }
+    * def users = karate.filter(response.users, fun)
+
+    And assert karate.sizeOf(users) == 1
+    And match users[0].username contains consortiaSystemUserName
+
+    # 6. non-primary affiliation for shadow 'consortia-system-user' has been created in 'user_tenant' table in 'central_mod_consortia'
+    * call read(login) consortiaAdmin
+    * def queryParams = { userId: '#(consortiaSystemUserInUniversityId)' }
+    Given path 'consortia', consortiumId, 'user-tenants'
+    And params query = queryParams
+    And headers {'x-okapi-tenant':'#(tenant)', 'x-okapi-token':'#(okapitoken)'}
+    When method GET
+    Then status 200
+
+    * def fun = function(userTenant) {return  userTenant.tenantId == centralTenant }
+    * def userTenants = karate.filter(response.userTenants, fun)
+
+    And assert karate.sizeOf(userTenants) == 1
+    And match userTenants[0].username contains consortiaSystemUserName
+    And match userTenants[0].isPrimary == false
+
+    # 7. shadow 'consortia-system-user' of 'universityTenant' has empty permissions (in 'centralTenant')
+    * call read(login) consortiaAdmin
+    Given path 'perms/users'
+    And param query = 'userId=' + consortiaSystemUserInUniversityId
+    And headers {'x-okapi-tenant':'#(tenant)', 'x-okapi-token':'#(okapitoken)'}
+    When method GET
+    Then status 200
+
+    And match response.totalRecords == 1
+    And match response.permissionUsers[0].permissions == []
+
+  # There will be only one admin user, so need to add permissions of 'consortiaAdmin' to shadow 'consortiaAdmin'
+  Scenario: Add permissions of 'consortiaAdmin' to shadow 'consortiaAdmin'
+    # get permissions of 'consortiaAdmin'
+    * call read(login) consortiaAdmin
+    Given path 'perms/users'
+    And param query = 'userId=' + id
+    And headers {'x-okapi-tenant':'#(tenant)', 'x-okapi-token':'#(okapitoken)'}
+    When method GET
+    Then status 200
+
+    * def newPermissions = $.permissionUsers[0].permissions
+
+    # get permissions of shadow 'consortiaAdmin'
+    * call read(login) consortiaAdmin
+    Given path 'perms/users'
+    And param query = 'userId=' + id
+    And headers {'x-okapi-tenant':'#(universityTenant)', 'x-okapi-token':'#(okapitoken)'}
+    When method GET
+    Then status 200
+
+    # add required permissions to shadow user
+    * def permissionEntry = $.permissionUsers[0]
+    * def updatedPermissions = karate.append(newPermissions, permissionEntry.permissions)
+    And set permissionEntry.permissions = updatedPermissions
+
+    # update permissions of shadow 'consortiaAdmin'
+    Given path 'perms/users', permissionEntry.id
+    And headers {'x-okapi-tenant':'#(universityTenant)', 'x-okapi-token':'#(okapitoken)'}
+    And request permissionEntry
+    When method PUT
+    Then status 200
+
+    # pause
+    * call pause 70000
```

Comment:
do you think we need such big pause of 70 seconds here?

Reply:
Usually it takes 60 seconds to synchronize updated permissions. This pause was in @PutPermissions in initData.feature file and I am using this tagged call in three places. Currently I have removed pause from initial place and for adding permissions to 'consortiaAdmin' and 'universityUser1' I am using retry in +'consortium.feature' file Lines 6, 15, so it prevents from using one more 1 minute pause. 
('mod-consortia' has dependency on 'mod-authtoken' so I have to do set up admin with required permissions in two steps and updating part is taking to much time.)